converters/PGNConverter.cpp

```diff
@@ -64,12 +64,20 @@ void PGNConverter::playAlgebraicMove(Color currPlayer, std::string algebraicStri
         case PAWN_MOVE:
             getPieceOrPawnPosition(currPlayer, algebraicString, board, startingPosition, endingPosition);
             break;
+        case PAWN_PROMOTION: {
+            getPieceOrPawnPosition(currPlayer, algebraicString, board, startingPosition, endingPosition);
+            std::regex pawnPromotionRegEx("=[NBQR]");
+            std::smatch sm;
+            std::regex_search(algebraicString, sm, pawnPromotionRegEx);
+            pawnPromotionMove(startingPosition, endingPosition, sm.str(0), board);
+            break;
+        }
         default:
             // Throw an exception since the pgn format is invalid
             throw std::invalid_argument("The pgn passed in is invalid.");
     }
 
-    if (!board->movePiece(startingPosition, endingPosition, currPlayer)) {
+    if (moveType != PAWN_PROMOTION && !board->movePiece(startingPosition, endingPosition, currPlayer)) {
```

Comment:
Not sure if this really feels intuitive. I feel like movePiece should potentially be able to handle the promotion of a pawn, that way we don't need this special case for 'PAWN_PROMOTION' and we simply move the piece like normal + pass in an extra parameter letting the move know what promotion value we want.

Reply:
Let me know if you have any criticisms about this method of tackling the problem.

Reply:
Or even create a proxy method movePieceAndPromote.